osu.Game/Rulesets/Objects/SliderPathExtensions.cs

```diff
@@ -16,7 +16,7 @@ public static class SliderPathExtensions
         public static void SnapTo<THitObject>(this THitObject hitObject, IDistanceSnapProvider? snapProvider)
             where THitObject : HitObject, IHasPath
         {
-            hitObject.Path.ExpectedDistance.Value = snapProvider?.FindSnappedDistance(hitObject, (float)hitObject.Path.CalculatedDistance) ?? hitObject.Path.CalculatedDistance;
+            hitObject.Path.ExpectedDistance.Value = snapProvider?.FindSnappedDistance(hitObject, (float)hitObject.Path.CalculatedDistance) * hitObject.DifficultyControlPoint.SliderVelocity ?? hitObject.Path.CalculatedDistance;
```

Comment:
This needs to divide by `SliderVelocity` before passing `CalculatedDistance` into `FindSnappedDistance` as well. The slider does correctly snap, but now it sorta "snaps beyond" the control points, which is fixed by this:
```csharp
double calculatedDistance = hitObject.Path.CalculatedDistance;
double sliderVelocity = hitObject.DifficultyControlPoint.SliderVelocity;
hitObject.Path.ExpectedDistance.Value = snapProvider?.FindSnappedDistance(hitObject, (float)(calculatedDistance / sliderVelocity)) * sliderVelocity ?? calculatedDistance;
```

https://user-images.githubusercontent.com/46171121/205200544-2f207942-54e5-4b60-b226-698f0d27168b.mp4